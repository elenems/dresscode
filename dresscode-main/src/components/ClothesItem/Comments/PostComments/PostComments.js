import React from "react";
import PropTypes from "prop-types";

function PostComments(props) {
  return (
    <div>
      <span className="label-style">Add a coment</span>
      <div className="post-comment-form">
        <textarea onChange={props.handleChange} value={props.commentText} />
        <button className="primary-button" onClick={props.addComment}>
          Add a comment
        </button>
      </div>
    </div>
  );
}

PostComments.propTypes = {
  commentText: PropTypes.string,
  handleChange: PropTypes.func.isRequired,
  addComment: PropTypes.func.isRequired
};

export default React.memo(PostComments);
