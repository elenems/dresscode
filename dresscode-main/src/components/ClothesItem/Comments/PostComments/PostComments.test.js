import React, {Fragment} from "react";
import { shallow } from "enzyme";
import PostComments from "./PostComments";

describe("PostComments", () => {
  it("Matches snapshot", () => {
    const wrapper = shallow(<PostComments />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("post-comment-form contains textarea and button", () => {
    const props = {
        commentText:'text',
        addComment:()=>{},
        handleChange:()=>{}
    }
    const wrapper = shallow(<PostComments {...props}/>);
    expect(wrapper.findWhere(n=>n.hasClass('post-comment-form')).children().get(0).type).toBe('textarea');
    expect(wrapper.findWhere(n=>n.hasClass('post-comment-form')).children().get(1).type).toBe('button');

  });
});
