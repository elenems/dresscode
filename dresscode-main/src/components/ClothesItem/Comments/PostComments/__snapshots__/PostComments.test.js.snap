// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostComments Matches snapshot 1`] = `
"<div>
  <span className=\\"label-style\\">
    Add a coment
  </span>
  <div className=\\"post-comment-form\\">
    <textarea onChange={[undefined]} value={[undefined]} />
    <button className=\\"primary-button\\" onClick={[undefined]}>
      Add a comment
    </button>
  </div>
</div>"
`;
